### DISCUSSION TITLE
Return a data frame with a path column instead of a list of paths for format = "file"

### DISCUSSION THREAD

**Discussion by programLyrique** [ropensci/targets#730] [Ideas]
I am writing a pipeline that has a target with branches, where each branch runs an external program that generate some files.
I want the target to track the external output files so I add `format = "file"`. 

However, I would also like to return some metadata about the run of the external program, for instance, the status code it returns, or the particular parameters passed to that program. 

I can separate that into two targets, one that generates the data frame with the metadata and the path, and a child one, which has `format = "file"` , grasps the path column from the data frame and just tracks the file paths. But it makes it less flexible to use the metadata later in the pipeline.

More generally, maybe it would be convenient to have a way to both track external files and some data handled directly in _targets at the same time for one target. What about being able to return a data frame with these data and a path column for a target with `format = "file"`?

**Comment by wlandau** [comment]
Sorry, but I do not have plans to support this. I see what you mean, but it is not a good fit for how `targets` works internally.

**Reply by programLyrique** [reply]
Thanks.

So I am thinking about this workaround: I can myself take care of the serialization of the data frame with metadata, save it as RDS or something like that, and add it to the list of paths the target returns. It's a bit hacky but it gives both tracking of the generated files and of the metadata.
